src/Data/ProgrammesDb/EntityRepository/PromotionRepository.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+namespace BBC\ProgrammesPagesService\Data\ProgrammesDb\EntityRepository;
+
+use BBC\ProgrammesPagesService\Domain\ValueObject\Pid;
+use DateTimeImmutable;
+use Doctrine\ORM\EntityRepository;
+use Doctrine\ORM\Query;
+
+class PromotionRepository extends EntityRepository
+{
+    /**
+     * @return array[]
+     */
+    public function findActivePromotionsByPid(Pid $pid, DateTimeImmutable $datetime, ?int $limit, int $offset): array
+    {
+        $qb = $this->createQueryBuilder('promotion')
+            ->addSelect(['context', 'promotionOfCoreEntity', 'promotionOfImage'])
```

Comment:
Also do we need the image related to the promotionOfCoreEntity so we have an image to display? If we need the image then we'll also need the ancestry of the coreEntity as the image may cascade down from one of it's parents.